java/clients/src/test/java/sleeper/clients/status/update/RenameTableIT.java

```diff
@@ -0,0 +1,152 @@
+import sleeper.configuration.properties.table.TablePropertiesStore;
+import sleeper.configuration.table.index.DynamoDBTableIndex;
+import sleeper.configuration.table.index.DynamoDBTableIndexCreator;
+import sleeper.core.CommonTestConstants;
+import sleeper.core.schema.Schema;
+import sleeper.core.table.TableIdentity;
+import sleeper.core.table.TableIndex;
+import sleeper.core.table.TableNotFoundException;
+import sleeper.core.table.TableWithNameAlreadyExistsException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static sleeper.configuration.properties.InstancePropertiesTestHelper.createTestInstanceProperties;
+import static sleeper.configuration.properties.instance.CdkDefinedInstanceProperty.CONFIG_BUCKET;
+import static sleeper.configuration.properties.table.TablePropertiesTestHelper.createTestTableProperties;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_ID;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.configuration.testutils.LocalStackAwsV1ClientHelper.buildAwsV1Client;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+import static sleeper.core.table.TableIdentity.uniqueIdAndName;
+
+@Testcontainers
+public class RenameTableIT {
+    @Container
+    public static LocalStackContainer localStackContainer = new LocalStackContainer(DockerImageName.parse(CommonTestConstants.LOCALSTACK_DOCKER_IMAGE))
+            .withServices(LocalStackContainer.Service.S3, LocalStackContainer.Service.DYNAMODB);
+
+    private final AmazonS3 s3 = buildAwsV1Client(localStackContainer, LocalStackContainer.Service.S3, AmazonS3ClientBuilder.standard());
+    private final AmazonDynamoDB dynamoDB = buildAwsV1Client(localStackContainer, LocalStackContainer.Service.S3, AmazonDynamoDBClientBuilder.standard());
+    private final InstanceProperties instanceProperties = createTestInstanceProperties();
+    private final Schema schema = schemaWithKey("key1");
+    private final TableIndex tableIndex = new DynamoDBTableIndex(instanceProperties, dynamoDB);
+    private final TablePropertiesStore propertiesStore = S3TableProperties.getStore(instanceProperties, s3, dynamoDB);
+
+    @BeforeEach
+    void setUp() {
+        s3.createBucket(instanceProperties.get(CONFIG_BUCKET));
+        DynamoDBTableIndexCreator.create(dynamoDB, instanceProperties);
+    }
+
+    @Test
+    void shouldRenameExistingTableUsingTableIdentity() {
+        // Given
+        TableIdentity oldTableIdentity = uniqueIdAndName("table-1-id", "old-name");
+        TableProperties oldProperties = createTable(oldTableIdentity);
+
+        // When
+        renameTable(oldTableIdentity, "new-name");
+
+        // Then
+        TableProperties expectedProperties = TableProperties.copyOf(oldProperties);
+        expectedProperties.set(TABLE_NAME, "new-name");
```

Comment:
This is copying TableProperties after the production code is invoked. The production code may have mutated the TableProperties object, so this isn't really asserting on what it looks like. The other tests do this as well.